.github/workflows/cabal.project.ghc-lib

```diff
@@ -0,0 +1,7 @@
+package ghc-tags
```

Comment:
These can't be here, because they are not considered when people issue `cabal install ghc-tags` on their computers. It needs to work without these annotations.

Same for other .project files.

Reply:
That's good point.  Unfortunately here doesn't seem to be a way to define flags in a `.cabal` file.  I think the only option is to:

* make `ghc-tags-core` build with `ghc-lib` by default
* support `ghc-lib` for `ghc-tags` by default as well, and only only non `ghc-lib` base build if one specifically asks for it (rather than chose `ghc` if the compiler version is the right one)

Would that be fine? Is there a better option?

Reply:
> This will not work for ghc-tags-plugin though, it must be build against ghc not ghc-lib.

Yes, that is indeed quite an unfortunate situation.

> The benefit is quicker builds for all GHC versions (as the ghc library is pre-installed).

I prefer the benefit of dealing with only one version of the GHC API, thus not drowning in CPP and ceasing maintenance once I've had enough ;) Dealing with GHC API breakages with every major release irritates me enough to be honest.

I wonder if the best course of action for now would be to leave things as-is. The duplication of code is somewhat unfortunate, but we can backport changes needed for new GHC versions from each other anyway :thinking:

Reply:
> I prefer the benefit of dealing with only one version of the GHC API, thus not drowning in CPP and ceasing maintenance once I've had enough ;) Dealing with GHC API breakages with every major release irritates me enough to be honest.

There's only initial price to pay, supporting new version with CPP is only slightly more annoying than supporting only the new version.

I don't even mind to maintain a fork of `ghc-tags`.